src/webui/abstractwebapplication.cpp

```diff
@@ -325,17 +274,31 @@ bool AbstractWebApplication::isAuthNeeded()
             || Preferences::instance()->isWebUiLocalAuthEnabled();
 }
 
-void AbstractWebApplication::printFile(const QString& path)
+int AbstractWebApplication::maxAllowedFilesize() const
 {
-    QByteArray data;
-    QString type;
+    return MAX_ALLOWED_FILESIZE;
+}
 
-    if (!readFile(path, data, type)) {
+bool AbstractWebApplication::printFile(const QString &path)
+{
+    QFile file {path};
+    if (!file.open(QIODevice::ReadOnly)) {
+        qDebug("File %s was not found!", qUtf8Printable(path));
         status(404, "Not Found");
-        return;
+        return false;
+    }
+    
+    if (file.size() > maxAllowedFilesize()) {
+        qWarning("%s: exceeded the maximum allowed file size!", qUtf8Printable(path));
+        status(500, "Internal Server Error");
```

Comment:
AFAIR, this was already asked by @Chocobo1, but I can't find that discussion. So I repeat, sorry: why  not 413 status code (Payload Too Large)?

Reply:
No I didn't and quoting RFC 7231:
>The 413 (Payload Too Large) status code indicates that the server is
   refusing to process a request because the request payload is larger
   than the server is willing or able to process

I.e. the request is too large not the response.

Reply:
Thanks, @Chocobo1!

Reply:
you're welcome!